File: Python/665. Non-decreasing Array.py
class Solution:
    def checkPossibility(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        ans = float('inf')
        for i in range(len(nums) - 1):
            if nums[i] > nums[i+1]:
                if ans != float('inf'):
                    return False
                ans = i
        if ans == float('inf') or ans == 0:
            return True
        if ans == len(nums) - 2 or nums[ans-1] <= nums[ans+1]:
            return True
        if nums[ans] <= nums[ans+2]:
            return True
        return False

File: C++/665. Non-decreasing Array.cpp
class Solution {
public:
    bool checkPossibility(vector<int>& nums) {
        int ans = INT_MAX;
        for(int i=0;i<nums.size()-1;i++)
        {
            if(nums[i] > nums[i+1])
            {
                if(ans != INT_MAX)
                    return false;
                ans = i;
            }
        }
        if(ans == INT_MAX || ans == 0 || ans == nums.size()-2)
            return true;
        else if(nums[ans-1] <= nums[ans+1] 
                || nums[ans] <= nums[ans+2])
            return true;
        return false;
            
    }
};